lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -192,15 +193,19 @@ def _check_workspace_permissions(self, wsid):
                 )
 
     def _check_workspace_permissions_list(self, wsids):
-        perms = self.sdkmr.get_workspace_auth().can_write_list(wsids)
-        bad_ws = [key for key in perms.keys() if perms[key] is False]
-        if bad_ws:
-            self.logger.debug(
-                f"User {self.sdkmr.user_id} doesn't have permission to run jobs in workspace {bad_ws}."
-            )
-            raise PermissionError(
-                f"User {self.sdkmr.user_id} doesn't have permission to run jobs in workspace {bad_ws}."
+        unique_not_none_not_zero_wsids = [wsid for wsid in set(wsids) if wsid]
+        if unique_not_none_not_zero_wsids:
+            perms = self.sdkmr.get_workspace_auth().can_write_list(
+                unique_not_none_not_zero_wsids
             )
+            bad_ws = [key for key in perms.keys() if perms[key] is False]
```

Comment:
Is there a reason for `is False` instead of `not perms[key]`?

Reply:
The function returns "True" or "False" so it explicitly checks against that.  Seems a bit more readable too.

Reply:
Fair enough